### PR TITLE
Fix usage of temporary string

### LINKED ISSUE #173: Mismatched operator new[] and delete
**Describe the bug**

[`wchar_t* temp`](https://github.com/Microsoft/calculator/blob/057401f5f2b4bb1ea143da02c773ac18d1bb9a2e/src/CalcViewModel/StandardCalculatorViewModel.cpp#L480) is [initialized with `new[]`, but deleted with `delete`](https://docs.microsoft.com/en-us/visualstudio/code-quality/c6283?view=vs-2017).

**Steps To Reproduce**
I just enabled code analysis. C'mon, Microsoft, it's your own tool!

**Additional context**

### PR DESCRIPTION
**Pull request by janisozaur**
Fixes #173

### CODE REVIEW

**Comment by janisozaur** [comment]
It reads `Build #0.1903.767.0 has failed`, but I can see no error. Can someone please provide information on where it failed?

**Comment by janisozaur** [comment]
Clicking around the linked page revealed it might be the infamous

> The agent: essential000002 lost communication with the server. Verify the machine is running and has a healthy network connection. For more information, see: https://go.microsoft.com/fwlink/?linkid=846610

I will rebase this commit to re-trigger the CI.

**Comment by mcooley** [comment]
We're still tweaking the settings on the pool of machines which run unit tests (essential##). Please run the tests locally to make sure they pass--we'll take care of re-queuing the CI tests on your behalf.

File: src/CalcViewModel/StandardCalculatorViewModel.cpp [hunk 1]
```diff
@@ -488,7 +488,7 @@ void StandardCalculatorViewModel::HandleUpdatedOperandData(Command cmdenum)
         m_selectedExpressionLastData = L"";
         if (ch == 'x')
         {
-            temp = L'\0';
+            temp[0] = L'\0';
```
Comment (jlaanstra) [inline src/CalcViewModel/StandardCalculatorViewModel.cpp]: While your're changing this, let's remove the new and replace it with something better.

**Pull request merged**